### ISSUE TITLE
Update librkllmrt.so library to version 1.2.1

### ISSUE THREAD

**Issue by alefris** [c0zaut/RKLLM-Gradio#16]
Hi.
Would you be able to bump up the library? version 1.2.1 is out: https://github.com/airockchip/rknn-llm/raw/refs/heads/main/rkllm-runtime/Linux/librkllm_api/aarch64/librkllmrt.so

I tried to simply replace the new file in the lib folder. 
I can launch the gradio interface, I can load a model but as soon as I submit my message, I get a segmentation fault. I am not really able to debug from there. I used dgb and faulthandler and seems the error has to do with the glibc library.
Perhaps the ctypes_bindings.py needs to be updated?

Thank you very much

**Comment by alefris** [comment]
I actually figured out how to make it work with library 1.2.1. What is needed is this:

1) in File **ctypes_bindings.py** 

1.a) in class _class RKLLMParam(ctypes.Structure):_
add the line:
        _("n_keep", ctypes.c_int32),_

1.b) modify the class _RKLLMExtendParam(ctypes.Structure):_ to look like this:
    fields_ = [
        ("base_domain_id", ctypes.c_int32),
        ("embed_flash", ctypes.c_int8),
        ("enabled_cpus_num", ctypes.c_int8),
        ("enabled_cpus_mask", ctypes.c_uint32),
        ("reserved", ctypes.c_uint8 * 106)
    ]

----- OPTIONAL (not needed but does not do harm) -----

1.c) after the line _RKLLMInferMode.RKLLM_INFER_GET_LAST_HIDDEN_LAYER = 1_
add the line 
_RKLLMInferMode.RKLLM_INFER_GET_LOGITS = 2_


1.d) modify the class _class RKLLMResult(ctypes.Structure):_ to look like this:
    _fields_ = [
        ("text", ctypes.c_char_p),
        ("token_id", ctypes.c_int),
        ("last_hidden_layer", RKLLMResultLastHiddenLayer),
        ("logits", RKLLMResultLogits)
    ]

1.e) add this class:
class RKLLMResultLogits(ctypes.Structure):
    _fields_ = [
        ("logits", ctypes.POINTER(ctypes.c_float)),
        ("vocab_size", ctypes.c_int),
        ("num_tokens", ctypes.c_int)
    ]

------------------------------------------------------------------------------------------------

2) in file  **model_class.py**

2.a) in class _RKLLMLoaderClass_, under the line _self.rkllm_param.extend_param.base_domain_id = 0_ add the lines:
        _self.rkllm_param.extend_param.enabled_cpus_num = 4
        self.rkllm_param.extend_param.enabled_cpus_mask = (1 << 4)|(1 << 5)|(1 << 6)|(1 << 7)_

2.b) in class RKLLMLoaderClass:
add the line:
 _self.rkllm_param.n_keep = -1_
